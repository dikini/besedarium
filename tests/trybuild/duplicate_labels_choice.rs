use besedarium::*;

struct L1; impl ProtocolLabel for L1 {}
struct L2; impl ProtocolLabel for L2 {}

// This should fail: duplicate label L1
// (Trybuild will check that this fails to compile)
type DuplicateLabels = TChoice<
    Http,
    L1,
    TInteract<Http, L1, TClient, Message, TEnd<Http, EmptyLabel>>,
    TInteract<Http, L1, TServer, Response, TEnd<Http, EmptyLabel>>
>;
assert_unique_labels!(DuplicateLabels);