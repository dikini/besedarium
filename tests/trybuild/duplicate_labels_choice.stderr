error[E0601]: `main` function not found in crate `$CRATE`
  --> tests/trybuild/duplicate_labels_choice.rs:14:40
   |
14 | assert_unique_labels!(DuplicateLabels);
   |                                        ^ consider adding a `main` function to `$DIR/tests/trybuild/duplicate_labels_choice.rs`
